packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/record-crud/record-crud.workflow-action.ts

```diff
@@ -40,6 +67,164 @@ export class RecordCRUDWorkflowAction implements WorkflowAction {
 
     const createdObjectRecord = await repository.save(objectRecord);
 
-    return { result: createdObjectRecord };
+    return {
+      result: createdObjectRecord,
+    };
+  }
+
+  private async updateRecord(
+    workflowActionInput: WorkflowUpdateRecordActionInput,
+  ): Promise<WorkflowActionResult> {
+    const repository = await this.twentyORMManager.getRepository(
+      workflowActionInput.objectName,
+    );
+
+    const objectRecord = await repository.findOne({
+      where: {
+        id: workflowActionInput.objectRecordId,
+      },
+    });
+
+    if (!objectRecord) {
+      throw new RecordCRUDActionException(
+        `Record ${workflowActionInput.objectName} with id ${workflowActionInput.objectRecordId} not found`,
+        RecordCRUDActionExceptionCode.RECORD_NOT_FOUND,
+      );
+    }
+
+    const updatedObjectRecord = await repository.update(
+      workflowActionInput.objectRecordId,
+      {
+        ...workflowActionInput.objectRecord,
+      },
+    );
+
+    return {
+      result: updatedObjectRecord,
+    };
+  }
+
+  private async deleteRecord(
+    workflowActionInput: WorkflowDeleteRecordActionInput,
+  ): Promise<WorkflowActionResult> {
+    const repository = await this.twentyORMManager.getRepository(
+      workflowActionInput.objectName,
+    );
+
+    const objectRecord = await repository.findOne({
+      where: {
+        id: workflowActionInput.objectRecordId,
+      },
+    });
+
+    if (!objectRecord) {
+      throw new RecordCRUDActionException(
+        `Record ${workflowActionInput.objectName} with id ${workflowActionInput.objectRecordId} not found`,
+        RecordCRUDActionExceptionCode.RECORD_NOT_FOUND,
+      );
+    }
+
+    const deletedObjectRecord = await repository.remove(objectRecord);
```

Comment:
are you sure about `.remove`? `.delete` instead?

Reply:
I replaced with a soft delete